src/Popper.jsx

```diff
@@ -38,15 +39,26 @@ class Popper extends Component {
     }
   }
 
-  componentDidUpdate(lastProps) {
+  componentDidUpdate(lastProps, lastState) {
+    const { children, eventsEnabled, onFlip, placement } = this.props
+    const { data: { placement: currentPlacement, flipped } } = this.state
```

Comment:
If `data` is `undefined` or `null` we could have a `TypeError`
You can use `const { data: { placement: currentPlacement, flipped } = {} } = this.state`
or init `data` in the state object.